cmd/template/cluster/provider/common.go

```diff
@@ -83,13 +88,13 @@ func runMutation(ctx context.Context, client k8sclient.Interface, templateData i
 		return microerror.Mask(err)
 	}
 
-	for _, t := range input {
+	for _, t := range templates {
 		// Add separators to make the entire file valid yaml and allow easy appending.
 		_, err = output.Write([]byte("---\n"))
 		if err != nil {
 			return microerror.Mask(err)
 		}
-		te := template.Must(template.New("resource").Parse(t))
+		te := template.Must(template.New(t.Name).Parse(t.Data))
```

Comment:
Wit this it prints errors like:

```
panic: template: kubeadm_control_plane.yaml.tmpl:73: function "ds" not defined
```

instead of:

```
panic: template: resource:73: function "ds" not defined
```

Reply:
Again my fault:
https://github.com/giantswarm/kubectl-gs/pull/479/files#diff-8b745f549ca7ff169a1b227062c187b5179abf71447c8fcd7fe0c5c885f17882R20

Needs to be escaped because CAPI uses go templating as well - see here:
https://github.com/giantswarm/kubectl-gs/blob/2007b08e8a7fc42d56be7abac0bd96aa4bf7267b/cmd/template/cluster/provider/templates/aws/kubeadm_control_plane.yaml.tmpl#L48